File: spreevy/src/main/java/com/stoyanov/spreevy/model/Tutorial.java
package com.stoyanov.spreevy.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@Entity
@Table(name = "tutorials")
public class Tutorial {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "published")
    private boolean published;

    public Tutorial() {
    }

    public Tutorial(String title, String description, boolean published) {
        this.title = title;
        this.description = description;
        this.published = published;
    }

    @Override
    public String toString() {
        return "Tutorial [id=" + id + ", title=" + title + ", desc=" + description + ", published=" + published + "]";
    }
}

File: vue-spreevy/src/types/Tutorial.ts
interface TutorialAdd {
    id: null | string | number,
    title: string,
    description: string | string[] | number
}

interface TutorialView {
    id: string | number,
    title: string,
    description: string | string[] | number
}

export type { TutorialAdd, TutorialView }